File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/UserCommandsModule.java
package ro.razvan.java.dagger.atm;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;
import dagger.multibindings.StringKey;

@Module
public interface UserCommandsModule {

    @Binds
    @IntoMap
    @StringKey(DepositCommand.COMMAND_KEY_DEPOSIT)
    Command bindDepositCommand(DepositCommand command);

    @Binds
    @IntoMap
    @StringKey(WithdrawCommand.COMMAND_KEY_WITHDRAW)
    Command bindWithdrawCommand(WithdrawCommand command);

    @Binds
    @IntoMap
    @StringKey(BalanceCommand.COMMAND_KEY_BALANCE)
    Command bindBalanceCommand(BalanceCommand command);

    @Binds
    @IntoMap
    @StringKey(LogoutCommand.COMMAND_KEY_LOGOUT)
    Command bindLogoutCommand(LogoutCommand command);

}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/UserCommandsModule.kt
package ro.razvan.kotlin.dagger.atm

import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap
import dagger.multibindings.StringKey

@Module
interface UserCommandsModule {

    @Binds
    @IntoMap
    @StringKey(COMMAND_KEY_DEPOSIT)
    fun bindDepositCommand(command: DepositCommand): Command

    @Binds
    @IntoMap
    @StringKey(COMMAND_KEY_WITHDRAW)
    fun bindWithdrawCommand(command: WithdrawCommand): Command

}